interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterImpl.java

```diff
@@ -258,21 +256,23 @@ private static Boolean needsQuotes(Shape anyShape) {
     return !(NUMBER_DATA_TYPES.contains(dataType) || BOOLEAN_DATA_TYPE.equals(dataType));
   }
 
-  private static boolean hasAnArrayVariant(UnionShape unionShape) {
-    boolean hasAnArrayVariant = false;
+  private static boolean hasShape(Shape shape, Class<?> shapeClass) {
+    return shapeClass.isInstance(shape) ||
+        shape instanceof UnionShape && containsShape((UnionShape) shape, shapeClass);
+  }
+
+  private static boolean containsShape(UnionShape unionShape, Class<?> shapeClass) {
     for (Shape shape : unionShape.anyOf()) {
-      if (shape instanceof ArrayShape) {
-        hasAnArrayVariant = true;
-      } else if (!(shape instanceof NilShape)) {
-        return false;
```

Comment:
I truly don't remember why this was here

Reply:
what I understand is that it was used for checking that the the union is of array shapes only (including nil), so if there was a shape that was not an array or nil, then you stop the search and return false.

Reply:
That being said, I don't think this code does the same, because if it is a union of an ArrayShape and a NodeShape, it will return true anyway because matches "any match", right?

Reply:
The original idea of `hasAnArrayVariant` was that:

```java
// Let's imagine we have a parseType function

// Null doesn't have an array variant
assertFalse(hasAnArrayVariant(parseType("nil")));
assertFalse(hasAnArrayVariant(parseType("nil | nil")));
assertFalse(hasAnArrayVariant(parseType("nil | nil | null")));

// Unions of arrays have an array variant
assertTrue(hasAnArrayVariant(parseType("String[] | Integer[]")));
assertTrue(hasAnArrayVariant(parseType("Integer[] | String[]")));

// Null is ignored in unions of arrays
assertTrue(hasAnArrayVariant(parseType("String[] | Integer[] | nil")));
assertTrue(hasAnArrayVariant(parseType("String[] | nil | Integer[]")));
assertTrue(hasAnArrayVariant(parseType("nil | String[] | Integer[]")));
assertTrue(hasAnArrayVariant(parseType("Integer[] | String[] | nil")));
assertTrue(hasAnArrayVariant(parseType("Integer[] | nil | String[]")));
assertTrue(hasAnArrayVariant(parseType("nil | Integer[] | String[]")));

// Any other type is not ignored in unions of arrays
assertFalse(hasAnArrayVariant(parseType("string | Integer[] | String[]")));
assertFalse(hasAnArrayVariant(parseType("Integer[] | string | String[]")));
assertFalse(hasAnArrayVariant(parseType("Integer[] | String[] | string")));
```

### Why was it like that?
#### I don't remember
##### I'll try to remember 🤷

Reply:
I think it was related to which parameters may or may not be repeatable.

I would test against a parameters with type `Integer | String[]`, `Integer | Integer[]` and `String[] | Integer`, etc and see which validations they trigger (or don't trigger) when passed on query params, json bodies and form-data.

Also... a lot of work to try out all that 😪